pkg/configuration/toolchainconfig/configuration.go

```diff
@@ -0,0 +1,373 @@
+}
+
+func (r RegistrationServiceVerificationConfig) MessageTemplate() string {
+	return getString(r.c.MessageTemplate, "Developer Sandbox for Red Hat OpenShift: Your verification code is %s")
+}
+
+func (r RegistrationServiceVerificationConfig) ExcludedEmailDomains() string {
+	return getString(r.c.ExcludedEmailDomains, "")
+}
+
+func (r RegistrationServiceVerificationConfig) CodeExpiresInMin() int {
+	return getInt(r.c.CodeExpiresInMin, 5)
+}
+
+func (r RegistrationServiceVerificationConfig) TwilioAccountSID() string {
+	key := getString(r.c.Secret.TwilioAccountSID, "")
+	return r.registrationServiceSecret(key)
+}
+
+func (r RegistrationServiceVerificationConfig) TwilioAuthToken() string {
+	key := getString(r.c.Secret.TwilioAuthToken, "")
+	return r.registrationServiceSecret(key)
+}
+
+func (r RegistrationServiceVerificationConfig) TwilioFromNumber() string {
+	key := getString(r.c.Secret.TwilioFromNumber, "")
+	return r.registrationServiceSecret(key)
+}
+
+type TiersConfig struct {
+	tiers toolchainv1alpha1.TiersConfig
+}
+
+func (d TiersConfig) DefaultTier() string {
+	return getString(d.tiers.DefaultTier, "base")
+}
+
+func (d TiersConfig) DurationBeforeChangeTierRequestDeletion() time.Duration {
+	v := getString(d.tiers.DurationBeforeChangeTierRequestDeletion, "24h")
+	duration, err := time.ParseDuration(v)
+	if err != nil {
+		duration = 24 * time.Hour
+	}
+	return duration
+}
+
+func (d TiersConfig) TemplateUpdateRequestMaxPoolSize() int {
+	return getInt(d.tiers.TemplateUpdateRequestMaxPoolSize, 5)
+}
+
+type ToolchainStatusConfig struct {
+	t toolchainv1alpha1.ToolchainStatusConfig
+}
+
+func (d ToolchainStatusConfig) ToolchainStatusRefreshTime() time.Duration {
+	v := getString(d.t.ToolchainStatusRefreshTime, "5s")
+	duration, err := time.ParseDuration(v)
+	if err != nil {
+		duration = 5 * time.Second
+	}
+	return duration
+}
+
+type UsersConfig struct {
+	c toolchainv1alpha1.UsersConfig
+}
+
+func (d UsersConfig) MasterUserRecordUpdateFailureThreshold() int {
+	return getInt(d.c.MasterUserRecordUpdateFailureThreshold, 2) // default: allow 1 failure, try again and then give up if failed again
+}
+
+func (d UsersConfig) ForbiddenUsernamePrefixes() []string {
+	prefixes := getString(d.c.ForbiddenUsernamePrefixes, "openshift,kube,default,redhat,sandbox")
+	v := strings.FieldsFunc(prefixes, func(c rune) bool {
+		return c == ','
+	})
+	return v
+}
+
+func (d UsersConfig) ForbiddenUsernameSuffixes() []string {
+	suffixes := getString(d.c.ForbiddenUsernameSuffixes, "admin")
+	v := strings.FieldsFunc(suffixes, func(c rune) bool {
+		return c == ','
+	})
+	return v
+}
+
+func getBool(value *bool, defaultValue bool) bool {
+	if value != nil {
+		return *value
+	}
+	return defaultValue
+}
+
+func getInt(value *int, defaultValue int) int {
+	if value != nil {
+		return *value
+	}
+	return defaultValue
+}
+
+func getString(value *string, defaultValue string) string {
+	if value != nil {
+		return *value
+	}
+	return defaultValue
+}
```

Comment:
this is already in common if I'm not mistaken

Reply:
Yep, I had moved it to common when I did the member operator refactoring. I'll remove this too.